pkg/publisher/local/publisher.go

```diff
@@ -24,18 +25,30 @@ type Publisher struct {
 	server        *LocalPublisherServer
 }
 
-func NewLocalPublisher(ctx context.Context, directory string, host string, port int) *Publisher {
+func NewLocalPublisher(ctx context.Context, directory string, host string, port int) (*Publisher, error) {
 	p := &Publisher{
 		baseDirectory: directory,
-		host:          ResolveAddress(ctx, host),
-		port:          port,
-		urlPrefix:     fmt.Sprintf("http://%s:%d", host, port),
+		// TODO: this field is only written to, never read. It could be deleted.
+		host:      ResolveAddress(ctx, host),
```

Comment:
As the TODO suggests, this value is never used. I think this package is ripe for a refactor at somepoint, but outside of scope here.